crates/ibc/src/applications/transfer/acknowledgement.rs

```diff
@@ -1,7 +1,7 @@
 use core::fmt::{Display, Error as FmtError, Formatter};
 
 use super::error::TokenTransferError;
-use crate::core::ics26_routing::context::Acknowledgement as AckTrait;
+use crate::core::ics04_channel::msgs::acknowledgement::Acknowledgement as AcknowledgementBytes;
```

Comment:
Would it be much change if we differentiate the namings of the two same `Acknowledgment` type? looks confusing. The enum inside transfer module is more of acknowledgement status. maybe something like `AckStatus` would make more sense.

Reply:
You're right, it wasn't really a default value. [c5415e6](https://github.com/cosmos/ibc-rs/pull/364/commits/c5415e637fb112382d25d000075de8a7ec300a34)

Reply:
Yeah, I found it difficult to name acks in a clear way. The challenge is that from the perspective of ibc core, an ack is just opaque bytes (`AcknowledgementBytes`). Then, every IBC app will have its own definition of what these bytes mean (`Acknowledgement` in this file). Normally, I would want to call it e.g. `TokenTransferAck`, but that would violate the Rust naming convention where the module name shouldn't be included in the item names...

So in this context, what you're suggesting is that `status` would encode "applying structure to the opaque bytes". I'll see if I can't think of better names; otherwise sure we could go with that.

Reply:
Right. I think I'd prefer `Acknowledgement` for the opaque bytes, and `TokenTransferAcknowledgement` for ics20 ack. WDYT @Farhad-Shabani ?

Reply:
> Right. I think I'd prefer `Acknowledgement` for the opaque bytes, and `TokenTransferAcknowledgement` for ics20 ack. WDYT @Farhad-Shabani ?

Sounds good to me. btw, please also consider `TokenTransferAckStatus` (at your choice), since we have something similar [here](https://github.com/cosmos/ibc-rs/blob/903a4ff88bf7984e1e3c0fef72efae678273e220/crates/ibc/src/applications/transfer/events.rs#L77)

Reply:
IMO I think `TokenTransferAcknowledgement` and `TokenTransferAckStatus` should be renamed

Reply:
Alright 👍🏻

Reply:
[bf8643c](https://github.com/cosmos/ibc-rs/pull/364/commits/bf8643ce880fd6e95eab3e8b6c7e5b07f033bcdf)